library/std/src/sys/thread_local/os.rs

```diff
@@ -68,6 +68,7 @@ impl<T: 'static> Storage<T> {
     ///
     /// The resulting pointer may not be used after reentrant inialialization
     /// or thread destruction has occurred.
+    #[inline]
     pub fn get(&'static self, i: Option<&mut Option<T>>, f: impl FnOnce() -> T) -> *const T {
```

Comment:
While you're at it, I think it might be beneficial to inline the `ptr.addr() == 1` case into this function, as that might yield more optimized `LocalKey::with`s.